src/iceberg/sort_order.cc

```diff
@@ -38,10 +39,48 @@ int32_t SortOrder::order_id() const { return order_id_; }
 
 std::span<const SortField> SortOrder::fields() const { return fields_; }
 
+bool SortOrder::Satisfies(const SortOrder& other) const {
+  // any ordering satisfies an unsorted ordering
+  if (other.IsUnsorted()) {
+    return true;
+  }
+
+  // this ordering cannot satisfy an ordering with more sort fields
+  if (fields_.size() < other.fields().size()) {
+    return false;
+  }
+
+  // this ordering has either more or the same number of sort fields
+  for (const auto& [field, other_field] : std::views::zip(fields_, other.fields_)) {
+    if (!field.Satisfies(other_field)) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+bool SortOrder::SameOrder(const SortOrder& other) const {
+  if (fields_.size() != other.fields().size()) {
+    return false;
+  }
+
+  for (const auto& [field, other_field] : std::views::zip(fields_, other.fields())) {
+    if (!(field == other_field)) {
+      return false;
+    }
+  }
+
+  return true;
```

Comment:
```suggestion
  return fields_ == other.fields_;
```

Can we simplify it as above?